### PR TITLE
Add a basic blueprint to create WP Playground instances.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

This adds a step to the live branches build that generates a Wordpress Playground link for testing PRs. I left live branches in place because it may have use cases that are not covered by Playground and for now we can't have Playground install non-public plugins like smooth generator unless we serve them from an address with cors headers. These are things we can investigate later.

### How to test the changes in this Pull Request:

Click the link in the Playground comment! Profit!

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by samueljseay** [comment]
~This is working, but blocked by a bug in playground https://github.com/WordPress/wordpress-playground/issues/1332~

This is working now thanks to the [suggested workaround](https://github.com/WordPress/wordpress-playground/issues/1332#issuecomment-2093988437)!

File: .github/workflows/pr-build-live-branch.yml
Comment (samueljseay) [inline .github/workflows/pr-build-live-branch.yml:5]: I removed this for testing. I will revert this again once I have approval to merge.

File: .github/workflows/pr-build-live-branch.yml
Comment (jonathansadowski) [inline .github/workflows/pr-build-live-branch.yml]: Should we note somewhere in the comment below that the retention period exists, or is there something on the plugin proxy below that keeps the package available beyond the retention period?

**Review by jonathansadowski — COMMENTED** [review]
Thanks for tackling this @samueljseay, exciting stuff!  I wasn't actually able to get the Playground to work on this PR using the link in the comment (I just see "There has been a critical error on this website. Please check your site admin email inbox for instructions."), but the zip generated by this workflow change appears to be the one used by the Playground (based on "Download as .zip" from the additional actions menu).

I added some questions about whether we should update the comment, and whether or not the retention of the asset is a concern.

File: .github/workflows/scripts/generate-playground-blueprint.js
Comment (jonathansadowski) [inline .github/workflows/scripts/generate-playground-blueprint.js]: Should we be editing a current comment if the build was updated?

Comment (samueljseay) [reply .github/workflows/scripts/generate-playground-blueprint.js]: @jonathansadowski good point, originally the plugin artifact name was stable so there wasn't a need, but now we generate it based on run id, I'll fix this.

**Comment by samueljseay** [comment]
@jonathansadowski I am now noticing that some requests that happen on startup of the environment are having issues, specifically requests to woocommerce.com that don't have cors headers (which playground can't handle), (By the way you can check console to see those errors). I will see if an update against trunk helps. 
Screenshot 2024-05-07 at 10 29 26 AM

File: .github/workflows/pr-build-live-branch.yml
Comment (samueljseay) [reply .github/workflows/pr-build-live-branch.yml]: @jonathansadowski yes that's a good point. Should we increase retention too, maybe to 30 days or something? 7 days is probably too short, I think I was messing around with it originally because I had hoped we could use live branches assets but they don't supply cors headers.

File: .github/workflows/pr-build-live-branch.yml
Comment (samueljseay) [inline .github/workflows/pr-build-live-branch.yml:5]: I'll remove this once the PR is approved. Just using it to test.

**Comment by samueljseay** [comment]
@jonathansadowski can you take another look? last 2 generated links work well for me.

Let me know how you feel about the 30 day retention.

File: .github/workflows/scripts/generate-playground-blueprint.js
Comment (samueljseay) [reply .github/workflows/scripts/generate-playground-blueprint.js]: done here https://github.com/woocommerce/woocommerce/pull/46761/commits/cdd39ff81f5e00c8d9deb9780e23d6a27858369a

File: .github/workflows/pr-build-live-branch.yml
Comment (samueljseay) [reply .github/workflows/pr-build-live-branch.yml]: set to 30 days here https://github.com/woocommerce/woocommerce/pull/46761/commits/814f47c2aa42603b06fa9f08327093e099f6136c

**Review by jonathansadowski — APPROVED** [review]
@samueljseay thanks for the updates. I think the 30 day retention is probably fine, but we may want to raise that discussion elsewhere, just to be sure. Either way, I think that can happen post-merge.

The comment update portion looks good, thanks for updating that.

I'm still receiving an error (a different one than you noted) when testing with the Playground:

```
PHP Fatal error:  Uncaught TypeError: property_exists(): Argument #1 ($object_or_class) must be of type object|string, null given in /wordpress/wp-content/plugins/woocommerce/src/Internal/Admin/Onboarding/OnboardingThemes.php:114
```

but that error seems unrelated to the changes in this PR to enable playground support, so I'll leave it up to you whether or not to merge.

**Comment by samueljseay** [comment]
Thanks @jonathansadowski Thats interesting, I was having troubles with that class before updating this PR against trunk so that's odd. Maybe it is due to it trying to load from an API that doesn't have CORS in some cases. (My testing mostly involved skipping store setup)

I'll merge it for now and get people to try it out and see what they encounter and maybe we can try diagnose these specific issues later. We may need to be realistic that not everything can currently work within the playground due to CORS issues and find ways around those issues in future.

File: .github/workflows/scripts/generate-playground-blueprint.js
Comment (psealock) [inline .github/workflows/scripts/generate-playground-blueprint.js:33]: Is it possible to add the Beta tester here?

Comment (psealock) [reply .github/workflows/scripts/generate-playground-blueprint.js:33]: Oh, I see the comment now that we can't. Thanks this is fantastic!

Comment (samueljseay) [reply .github/workflows/scripts/generate-playground-blueprint.js:33]: @psealock I don't think it's something I can influence or work on right now, but I think we should set up like a basic AWS s3 bucket or something to serve up a few of these plugins like smooth generator with cors headers.

**Pull request merged**
